src/components/tasks/review/SingleTaskReview.js

```diff
@@ -1,29 +1,49 @@
 import MonacoEditor from 'react-monaco-editor';
 import {decodeBase64} from '../../../utils/hooks/Base64';
 import {MarkdownViewer} from '../../commons/MarkdownViewer';
-import {Card} from '@material-ui/core'
+import {Card, Typography} from '@material-ui/core'
+import { useState, useRef, useEffect } from 'react'
+import {SingleTestCase} from './SingleTestCase'
+import {SingleResult} from './SingleResult'
 
 export const SingleTaskReview = ({ task }) => {
     const { code, results, tests, description } = task
 
+    const decodedResults = JSON.parse(decodeBase64(decodeBase64(results)))
+    const decodedTestCases = JSON.parse(decodeBase64(tests))
+
+    const [height, setHeight] = useState(0)
+    const ref = useRef(null)
```

Comment:
Może głupie pytanie jak działa ref? Jak zmieni się coś w komponencie który przekazaliśmy niżej to tu też się zmieni?

Reply:
W docsach Reacta piszą, że useRef przechowuje mutolwany obiekt, którego właściwość current reprezentuje obecny node drzewa DOM, przechowywany przez ref jako wartość początkową. Ogólnie korzystamy tu z tego tylko po to, żeby zapiąć się referencją na obiekt zadania, zeby MonacoEditor miał tę samą wysokość. 

Jest to podpierdzielone ze stacka, ale ogólnie useRef może służyć właśnie jako  imperatywny dostęp do potomka dokumentu. Nie robimy funkcji callbackowej, żeby dostać tę wysokość po wyrenderowaniu, ale mamy sobie zapisany reprezentujący go obiekt. Ogólnie nie lubię odsyłać do docsow, ale może moje tłumaczenie jest skomplikowane / złe, więc tu masz docsy: https://pl.reactjs.org/docs/hooks-reference.html#useref